encoding.go

```diff
@@ -111,6 +111,39 @@ func MakeTypedEncoder(f interface{}) func(*Request) func(io.Writer) Encoder {
 	})
 }
 
+// MakeMultiEncoder takes pairs of arguments, where the first of the pair is a value that denotes a type and the second denotes an encoder.
+// The resulting encoder then uses the encoder of the pair with matching first element.
+// Example:
+//    e := MakeMultiEncoder(
+//    	"string", MakeTypedEncoder(func(req *Request, w io.Writer, str string) error {
```

Comment:
Shouldn't we be able to use reflection to get the type from the function signature, instead of manually writing the type?

Reply:
We do, we could also pass an empty string but this is more descriptive.